simulator/infra/memory/memory.h

```diff
@@ -46,6 +43,15 @@ class FuncMemory
         size_t memcpy_guest_to_host( Byte* dst, Addr src, size_t size) const;
         size_t memcpy_guest_to_host_noexcept( Byte* dst, Addr src, size_t size) const noexcept;
 
+        template<typename T, Endian endian> T read( Addr addr) const;
```

Comment:
Maybe it worth setting a default endiannes argument as LE? E.g.
```c++
template<typename T, Endian endian = Endian::little> T read( Addr addr) const;
```

Reply:
Why not BE?

Reply:
I messed up a bit, it's not needed.